ambry-router/src/main/java/com.github.ambry.router/GetBlobOperation.java

```diff
@@ -1051,7 +1057,6 @@ protected void maybeProcessCallbacks() {
                   decryptCallbackResultInfo.result.getDecryptedUserMetadata().array());
             }
             totalSize = decryptCallbackResultInfo.result.getDecryptedBlobContent().remaining();
-            chunkSize = totalSize;
```

Comment:
Similar to previous comment. Where the `chunkSize` is populated? I notice that `getKeysAndSizesAndOffsets` contains chunk size info, however, note that `filterChunkToRange` method directly uses variable `chunkSize`, which might not be initialized. Please double check code there.

Reply:
Also, we can discuss offline about the chunk size of a simple blob.

Reply:
It's ok since chunkSize is only used when calculating the end of data chunk for a byte range operation if that data chunk is not the last data chunk. For simple data blobs, the FirstChunk will always be the last data chunk so it's unused, and for metadata blobs the size of the FirstChunk is irrelevant.